### ISSUE TITLE
Container fails to start when using relative volume paths in docker-compose.yml

### ISSUE THREAD

**Issue by deadbeef84** [rancher/rancher#3702]
Rancher Version: 0.59.0

Docker Version: 1.9.1

OS: Ubuntu 15.10

Steps to Reproduce: 
```yaml
web:
  image: httpd
  volumes:
    - ./htdocs:/usr/local/apache2/htdocs
```
And then try to boot it up using `$ rancher-compose up -d`

Results: 
Fail to start container, with the following exception in rancher-server
```400 Client Error: Bad Request ("./htdocs includes invalid characters for a local volume name, only [a-zA-Z0-9][a-zA-Z0-9_.-] are allowed")```

Expected:
Successful start of container, interpreting the volume as a relative path instead of a name. This works as expected when using **docker-compose**.

**Comment by deniseschannon** [comment]
What would you expect the relative path to be? We won't be uploading directories on your laptop to every host.

**Comment by keras** [comment]
I understand that it might not be as straight forward as it sounds. But how about the same thing than with the with `build: foo` jobs. Upload to s3 with the path being relative to docker-compose.yml.

P.S. Also it would be great if s3 wouldn't be required, and maybe the Rancher server could do that job

**Comment by deadbeef84** [comment]
Sad to see the won't fix tag, but let me try to explain why I want this feature. My use case for this would mostly be for development environments, and I agree that it doesn't make much sense for remote servers.

It's practical for every developer to be able to easily run and test our projects which consist of a few rancher stacks. It's nice to be able to fully replicate the production setup. It's not very practical to use absolute paths for volumes as each developer may have a completely different setup. **docker-compose** allowed us to do this.

I'm fine with the paths being resolved client side (as that's also where the docker-compose.yml which it would be relative to would reside). The paths wouldn't make much sense when running on remote hosts, though it would still be helpful when running locally. I'm not entirely sure how docker-compose handles this for remote hosts.

**Comment by crfeliz** [comment]
@deadbeef84 You can interpolate the PWD environment variable to do this. Note that this means it would be relative to the working directory not the docker-compose.yml.

**Comment by lucaspottersky** [comment]
@crfeliz could you be more explicit? I'm facing the same issue.

@will-chan my `docker-compose.yml` uses something like: `build: http://www.bitbucket.com/myrepo.git`. This repository has this structure:
- src/
- docker-compose.yml
- Dockerfile1
- Dockerfile2

And I'm trying to mount "src" like so:
```
web:
  build: http://www.bitbucket.com/myrepo.git
  volumes:
    - ./src:/some/container/folder
```

What are your advices? 
How can I mount my source code folder into the container?

**Comment by deniseschannon** [comment]
This is resolved and fixed in v1.1.0-dev2-rc4 using rancher-compose v0.8.2-rc1. 

If I use 

```
web:
  image: httpd
  volumes:
    - ./htdocs:/usr/local/apache2/htdocs
```

The container will start up with the relative path set in the API.

**Comment by MichelDiz** [comment]
I would like more information on this. Is Rancher CLI ready to send entire paths from my local computer to the host / cloud (mean files like src, build, images, etc)?

Or do I have to send the files to the server or agent and prepare a folder for it? and then set the paths in Rancher Service.

I was thinking of running into a CI method with "Drone", however I need to change a lot on my dockerfile and docker-compose. It would be very good to be able to stream a file to the host.

Cheers
